src/App.tsx

```diff
@@ -1,19 +1,86 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Error } from './components/Error';
+import { FieldForSorting, Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [fieldForSorting, setFieldForSorting]
+    = useState<FieldForSorting>(FieldForSorting.All);
+  const [counterActiveTodos, setCounterActiveTodos] = useState(0);
+
+  const getTodosFromAPI = useCallback(async () => {
+    setIsError(false);
+    if (user) {
+      try {
+        const recMovie = await getTodos(user.id);
+
+        setTodos(recMovie);
+      } catch {
+        setIsError(true);
+      }
+    }
+
+    setTimeout(() => {
+      setIsError(false);
+    }, 3000);
+  }, []);
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    getTodosFromAPI();
+  }, []);
+
+  const closeError = useCallback(() => {
+    setIsError(false);
   }, []);
 
+  const selectFieldForSorting = useCallback((fieldForSort: FieldForSorting) => {
+    setFieldForSorting(fieldForSort);
+  }, [fieldForSorting]);
+
+  const countActiveTodos = useCallback(() => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setCounterActiveTodos(todos.length - completedTodos.length);
+  }, [todos]);
+
+  useEffect(() => {
+    const filteredTodos = todos.filter(todo => {
+      switch (fieldForSorting) {
+        case FieldForSorting.Active:
+          return !todo.completed;
+
+        case FieldForSorting.Completed:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+
+    setVisibleTodos(filteredTodos);
+    countActiveTodos();
```

Comment:
I don't think that we should save it to state, just save filtration result to constant using useMemo to avoid unnecessary filtration